Sources/ClientRuntime/Config/DefaultSDKRuntimeConfiguration.swift

```diff
@@ -85,16 +85,18 @@ public struct DefaultSDKRuntimeConfiguration<DefaultSDKRuntimeRetryStrategy: Ret
 // Exposing these as static properties/methods allows them to be used by custom config objects.
 public extension DefaultSDKRuntimeConfiguration {
 
-    /// The default HTTP client to use when none is configured
+    /// The default HTTP client for the target platform, configured with the supplied configuration.
     ///
-    /// Is the CRT HTTP client.
-    static var defaultHttpClientEngine: HttpClientEngine { CRTClientEngine() }
-
-    /// The default HTTP client with a specified timeout
-    ///
-    /// Is the CRT HTTP client.
-    static func httpClientEngineWithTimeout(timeoutMs: UInt32) -> HttpClientEngine {
-        return CRTClientEngine(config: CRTClientEngineConfig(connectTimeoutMs: timeoutMs))
+    /// - Parameter httpClientConfiguration: The configuration for the HTTP client.
+    /// - Returns: The `CRTClientEngine` client on Mac & Linux platforms, returns `URLSessionHttpClient` on non-Mac Apple platforms.
+    static func makeClient(httpClientConfiguration: HttpClientConfiguration) -> HTTPClient {
+        #if os(iOS) || os(tvOS) || os(watchOS) || os(visionOS)
+        return URLSessionHTTPClient(httpClientConfiguration: httpClientConfiguration)
+        #else
+        let connectTimeoutMs = httpClientConfiguration.connectTimeout.map { UInt32($0 * 1_000_000) }
+        let config = CRTClientEngineConfig(connectTimeoutMs: connectTimeoutMs)
+        return CRTClientEngine(config: config)
+        #endif
```

Comment:
For now I have defaulted non-Mac Apple platforms to the new HTTP client because the CRT client is known to not work there.

In the long term, it should be possible to use either client on the Mac platform, but I don't want to provide an interface for doing so when SRA plugins is right around the corner.